js/widgets/pitchslider.js

```diff
@@ -44,27 +46,31 @@ function PitchSlider() {
         this._logo.blocks.loadNewBlocks(newStack);
     };
 
-    this.init = (logo) => {
-        if (window.widgetWindows.openWindows["slider"]) return;
-        if (!this.frequencies || !this.frequencies.length) this.frequencies = [392];
-        this._logo = logo;
+    init(logo){
```

Comment:
We don't need an `init`. `constructor` should do what `init` is meant to do. There's just one instance of this class (in `blocks/WidgetBlocks.js`), and only one `init` in the same scope. Remove the `new PitchSlider()` code. Replace the `init` call with `new PitchSlider()`. `logo` isn't required to be passed. It is _global_.

Reply:
I saw that one instance in block/WidgetBlocks.js. So you want me to remove init() and add entire code in constructor of Pitchslider? Or is it something else?

> We don't need an `init`. `constructor` should do what `init` is meant to do. There's just one instance of this class (in `blocks/WidgetBlocks.js`), and only one `init` in the same scope. Remove the `new PitchSlider()` code. Replace the `init` call with `new PitchSlider()`. `logo` isn't required to be passed. It is _global_.

Reply:
yes